### PR TITLE
Add end game states and flag states

### PR DESCRIPTION
**Pull request by CompSciLauren**
I added both end game states and the flag state.
It tracks total # of flags user has set and total # of correct flags.

EDIT: Just an FYI, the force-pushes are me removing random crap I added that showed up in the PR and should've been removed before opening this.

### CODE REVIEW

**Comment by CompSciLauren** [comment]
general comment: I realized that at some point days ago I had re-downloaded the project, and thus didn't have the dependencies installed anymore. The only dependency we have is Prettier I believe, but that's the one we said at the beginning of this project would be used to automatically fix the style of our code for us, so it would automatically be fixed whenever we commit (making the code look cleaner automatically). I just now installed it, so some of the changes are just my code being fixed.

Everyone should make sure they've run "npm install" to get the dependencies. This stuff is already on the readme I just forgot to do it again after I re-downloaded our project.

File: index.js [hunk 1]
```diff
@@ -47,14 +56,19 @@ function createArray(length) {
...
 
 //For storing values in the 2D array, used to create unique instance 'objects'. Can be called on by arr[x][y].<property>
-function squareProperties(isClicked, isBomb, isFlagged, numNeighborMines, testingClickTimes)
-{
+function squareProperties(
+  isClicked,
+  isBomb,
+  isFlagged,
+  numNeighborMines,
+  testingClickTimes
```
Comment (EricESeitz) [inline index.js]: testingClickTimes can be removed now. Not necessary for MVP but once we get close to a playable program these testing functions/parameters can be removed. Keep in mind

File: index.js [hunk 2]
```diff
@@ -170,71 +192,119 @@ function onClicked(x, y) {
...
+      //If the number of neighboring mines is 0, recurse. Otherwise stop
       recShowNonMineSquare(x, y);
       return;
     } else {
       return;
     }
+  } else if (arr[x][y].isBomb == 1 && arr[x][y].isFlagged == 0) {
+    alert("game over");
+    location.reload();
   } else {
```
Comment (EricESeitz) [inline index.js]: Is this else-if statement called correctly? There's an else statement above it that just returns

Comment (CompSciLauren) [reply index.js]: I think it's correct. That else statement above it is inside of the first if-statement. The out-most if/else set is just the "if", "else if", and "else", not including the one you mentioned.

File: index.js (hunk 1 above)
Comment (CompSciLauren) [reply index.js]: Went ahead and removed it!

**Comment by EricESeitz** [comment]
Approved, go ahead with the squash and merge

**Pull request merged**